saleor/dashboard/category/views.py

```diff
@@ -110,7 +112,10 @@ def category_delete(request, pk):
     if request.method == "POST":
         descendants = category.get_descendants()
         menus = get_menus_that_need_update(categories=descendants)
+        products = unpublished_products_before_category_delete(category)
+        product_ids = [product.pk for product in products]
         category.delete()
+        update_products_minimal_variant_prices_task.delay(product_ids)
```

Comment:
This four lines are executed every time when we delete a category. The ideal solution would be to put them into one function. I actually think that `unpublished_products_before_category_delete` could become the `delete_category` function that does all of that. It should also have a docstring which would tell you that this function removes a category plus it performs some additional logic related to unpublishing the products.

Reply:
New function: `delete_categories` in saleor/product/utils/\_\_init\__.py